jni_gen_test/tool/gen.dart

```diff
@@ -0,0 +1,24 @@
+import 'dart:io';
+import 'package:jni_gen/jni_gen.dart';
+
+void main(List<String> arguments) async {
```

Comment:
This should be a command-line interface in `jni_gen/bin/jni_gen.dart` so that we can do `dart run jni_gen --config <some-config-file>.yaml`.

For a test that uses the Dart API instead of a config file: We should in a test regenerate the file and check that it is equal to what we had before. (Give the `_test.dart` file a parameter to keep the temporary files, so we can easily inspect the API summary as well.)

Reply:
Is `golden file` testing viable at this point? Because generated code structure is still in flux.

Reply:
Yeah I think it's fine. It also makes it easy to see how we change our generated code on subsequent PRs.

Reply:
In `ffigen` I saw there was quite a bit of code dedicated for YAML parsing. Is there now something like `json_serializable` for YAML? I think YAML config would not be an immediate priority.

Reply:
We can leave the config for later.

In that case we will not have a `bin/jni_gen.dart` for now.

We will only have a `lib/jin_gen.dart` that exposes a Dart API that is directly used by our tests.